doc/adr/0002-use-component-level-micro-frontends-for-the-header-and-footer.md

```diff
@@ -0,0 +1,36 @@
+# 2. Use component-level micro frontends for the header and footer
+
+Date: 2020-03-25
+
+## Status
+
+Proposed
+
+## Context
+
+Having [separate micro frontends for pages] requires duplication of the header and footer.
+
+We expect users will want to change the content of both, as well as their styling and behaviour.
+
+## Decision
+
+We will create dedicated services for the header and footer.
+
+We will use [Edge Side Includes (ESIs)], limited to `<esi:include>`, to transclude them into pages. 
+
+Each component will have 2 ESIs: 1 for the `<head>` and 1 for the `<body>`.
+
+## Consequences
+
+- It is compatible with popular ESI processors (e.g. Varnish, Fastly).
+- Not all popular CDNs are ESI processors (e.g. AWS CloudFront). If there is no trusted proxy that can do so, each
+  page-level micro frontend will need to process the ESIs itself. They can detect this through the
+  [`Surrogate-Capability`][Surrogate-Capability] request header.
+- Components can add their own CSS and JavaScript in the `<head>` ESI.
+- As the page transcludes the component, developers and users will have to take care to avoid clashes in styling and
```

Comment:
Wouldn't the use of patterns and BEM help mitigate against this? Is it worth mentioning that? Granted others might extend it in ways that might break it, it seems like the consequences are minimal thanks to other efforts.

Reply:
They do, which is why we used them. There's no ADR to link to though.

Reply:
Fair point. Put that `patterns and BEM ADR` on the list ;)